sbi/inference/snpe/snpe_c.py

```diff
@@ -411,7 +411,11 @@ def _log_prob_proposal_posterior_mog(
         log_prob_proposal_posterior = utils.mog_log_prob(
             theta, logits_pp, m_pp, prec_pp
         )
-        utils.assert_all_finite(log_prob_proposal_posterior, "proposal posterior eval")
+        utils.assert_all_finite(
+            log_prob_proposal_posterior,
+            """the evaluation of the MoG proposal posterior. This is likely due to too
+            little training data---consider increasing your simulation budget.""",
+        )
```

Comment:
I'm not 100% sure about this -- are you still observing the failure with low simulation-budgets after the fix?